Frameworks/Foundation/NSString.mm

```diff
@@ -680,7 +680,25 @@ - (instancetype)pathExtension {
  @Status Interoperable
 */
 - (instancetype)lastPathComponent {
-    return [self substringFromIndex:_CFStartOfLastPathComponent2(static_cast<CFStringRef>(self))];
+    if ([self length] == 0) {
+        return @"";
+    }
+
+    CFIndex index = _CFStartOfLastPathComponent2(static_cast<CFStringRef>(self));
+    NSString* substring = [self substringFromIndex:index];
+
+    if (IS_SLASH([self characterAtIndex:index])) {
+        if ([self length] == 1) {
+            return substring;
+        }
+
+        // only way we could of ended up here is, if we have a string with multiple leading slashes.
+        NSArray<NSString*>* tokens = [self componentsSeparatedByString:_NSGetSlashStr()];
+        return ([tokens[0] isEqualToString:@""]) ? _NSGetSlashStr() : tokens[0];
```

Comment:
right, but if you ran this code on '/foo/bar/////', this would return 'foo' instead of 'bar'

---
In reply to: [74352571](https://github.com/Microsoft/WinObjC/pull/870#discussion_r74352571) [](ancestors = 74352571,74352319,74352126,74351965)